server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -490,6 +502,16 @@ private void updateCleanupKeyToCountMapOnCacheInsertion(CleanupKey cleanupKey) {
             cleanupKeyToCountMap.computeIfAbsent(shardId, k -> new HashMap<>()).merge(cleanupKey.readerCacheKeyId, 1, Integer::sum);
         }
 
+        /**
+         * Updates the cleanupKeyToCountMap and staleKeysCount when a cache eviction occurs.
+         *
+         * <p>This method is called when an entry is evicted from the cache.
+         * It decrements the count of the entry in the cleanupKeyToCountMap.
+         * It also decrements the staleKeysCount only if the entry was accounted.
+         * If the count of the CleanupKey becomes zero, it removes the CleanupKey from the map.
```

Comment:
lets fix the documentation to reflect that updates happen for any key removed from cached with reason other than replaced.